packages/core/src/shorthand-parser/index.ts

```diff
@@ -92,26 +96,77 @@ export const animation = createPropertyParser((_: any, css: any, value: any, ind
-  if (matchString(value, /^[0-9.]+deg/)) css.fontStyle += ` ${value}`;
-  else if (matchString(value, /\//)) {
-    const [fontSize, lineHeight] = value.split('/');
-    css.fontSize = tokens.fontSizes[fontSize] || fontSize;
-    css.lineHeight = tokens.lineHeights[lineHeight] || lineHeight;
-  } else if (
-    matchString(value, unitMatch) ||
-    matchString(value, /xx-small|x-small|small|medium|large|x-large|xx-large|xxx-large/) ||
-    tokens.fontSizes[value]
-  )
-    css.fontSize = tokens.fontSizes[value] || value;
-  else if (matchString(value, /normal|italic|oblique/)) css.fontStyle = value;
-  else if (matchString(value, /normal|bold/) || tokens.fontWeights[value])
-    css.fontWeight = value === 'normal' ? 400 : tokens.fontWeights[value] || 700;
-  else if (matchString(value, unitMatch) || tokens.fontSizes[value]) css.fontSize = tokens.fontSizes[value] || value;
-  else {
-    css.fontFamily = setChainedValue(css.fontFamily, tokens.fonts[value] || value);
+export const font = createPropertyParser(
+  (tokens: any, css: any, value: any, index: any, chain: any, chainIndex: any, chains: any) => {
+    if (chains.shouldParseFontFamily) {
+      css.fontFamily = setChainedValue(css.fontFamily, tokens.fonts[value] || value);
+    } else {
+      const lower = value.toLowerCase();
+
+      switch (true) {
+        case fontStyleMatch.test(lower):
+          css.fontStyle += ` ${value}`;
+
+          break;
+
+        case fontWeightMatch.test(lower):
+          css.fontWeight = tokens.fontWeights[value] || value;
+
+          break;
+
+        case fontSizeMatch.test(lower):
+          chains.shouldParseFontFamily = true;
+
+          const [fontSize, lineHeight] = value.split('/');
+          css.fontSize = tokens.fontSizes[fontSize] || fontSize;
+
+          if (lineHeight) {
+            css.lineHeight = tokens.lineHeights[lineHeight] || lineHeight;
+          }
+
+          break;
+
+        case lower === 'italic':
+        case lower === 'oblique':
+          css.fontStyle = value;
+
+          break;
+
+        case lower === 'small-caps':
+          css.fontVariant = value;
+
+          break;
+
+        case lower === 'condensed':
+        case lower === 'expanded':
+        case lower === 'extra-condensed':
+        case lower === 'extra-expanded':
+        case lower === 'semi-condensed':
+        case lower === 'semi-expanded':
+        case lower === 'ultra-condensed':
+        case lower === 'ultra-expanded':
+          css.fontStretch = value;
+
+          break;
+
+        case lower === 'caption':
+        case lower === 'icon':
+        case lower === 'menu':
+        case lower === 'message-box':
+        case lower === 'small-caption':
+        case lower === 'status-bar':
+          chains.shouldParseFontFamily = true;
+
+          css.fontFamily = setChainedValue(css.fontFamily, tokens.fonts[value] || value);
+
+          break;
+
+        default:
+          break;
```

Comment:
This was added during the rebase in order to respect an eslint/tslint rule which I had not encountered beforehand. An alternative that I considered was `// eslint-disable-next-line switch-default` over the initial `switch` statement, a technique I saw used elsewhere. However, the rule _exists_, and so it seemed better to include the `default` statement to be explicit (tho “noisey” to me), rather than immediately suggesting a change your linting configuration.

Reply:
Sounds good. we are going to be reviewing the lining rules at a later time but for now, it's a good idea to stitch with them.